File: src/main/java/yunas/ErrorController.java
package yunas;

import javax.servlet.http.HttpServletRequest;
import java.util.function.BiFunction;

/**
 * ErrorController Interface for Java.
 *
 */
public interface ErrorController extends BiFunction<HttpServletRequest, Integer,Object> {

    public Object action(HttpServletRequest req, Integer integer);

    @Override
    default Object apply(HttpServletRequest req, Integer integer) {
        return action(req, integer);

    }
}

File: src/main/java/yunas/ErrorController.kt
package yunas

import javax.servlet.http.HttpServletRequest
import java.util.function.BiFunction

/**
 * ErrorController Interface for Java.
 *
 */
@FunctionalInterface
interface ErrorController : BiFunction<HttpServletRequest, Int, Any> {

    fun action(req: HttpServletRequest, integer: Int?): Any

    override fun apply(req: HttpServletRequest, integer: Int): Any {
        return action(req, integer)

    }
}
